tests/integ/modin/index/test_reindex.py

```diff
@@ -0,0 +1,269 @@
+#
+# Copyright (c) 2012-2024 Snowflake Computing Inc. All rights reserved.
+#
+
+from typing import Callable
+
+import modin.pandas as pd
+import numpy as np
+import pandas as native_pd
+import pytest
+
+import snowflake.snowpark.modin.plugin  # noqa: F401
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from tests.integ.modin.sql_counter import SqlCounter, sql_count_checker
+from tests.integ.modin.utils import assert_index_equal
+
+
+def assert_reindex_result_equal(
+    snow_result: tuple[pd.Index, np.ndarray], native_result: tuple[pd.Index, np.ndarray]
+) -> None:
+    assert_index_equal(snow_result[0], native_result[0])
+    np.testing.assert_array_equal(snow_result[1], native_result[1])
+
+
+def assert_reindex_exceptions_match(
+    snow_idx: pd.Index,
+    native_idx: native_pd.Index,
+    func: Callable,
+    excp_type: Exception,
+    excp_match: str,
+) -> None:
+    with pytest.raises(excp_type, match=excp_match):
+        func(native_idx)
+
+    with pytest.raises(excp_type, match=excp_match):
+        func(snow_idx)
+
+
+def create_datetime_index(snow_series: pd.Series) -> pd.Index:
```

Comment:
Ideally we should not be manipulating Snowpark pandas internals in this test. I think we can delete this method and use `pd.Index(snow_series)` everywhere we need a datetime index.

Reply:
At commit c5c80043c49f455fbd79816dc36055b9eefb76eb, `pd.Index(pd.date_range("1/1/2010", periods=6, freq="D"))` is giving me a lazy `snowflake.snowpark.modin.plugin.extensions.index.Index` 

Alternatively we can do `pd.date_range("1/1/2010", periods=6, freq="D").to_frame().set_index(0)`, which seems to use fewer queries. Note that the name is 0 because for reasons I don't understand (maybe a pandas bug), `native_pd.Series().to_frame()` and `pd.Series().to_frame()` give the column name 0.

Reply:
Hmm - seems like the bug was fixed! I'll go ahead and remove this workaround!